Nodejs/Product/Nodejs/NodejsProject.cs

```diff
@@ -574,11 +572,16 @@ out frame
             return hr;
         }
 
+        private bool IsES6IntellisensePreview() {
+            // If the analysis level is set to preview then use the TypeScript language service for node.js
+            return NodejsPackage.Instance.IntellisenseOptionsPage.AnalysisLevel == Options.AnalysisLevel.Preview;
+        }
+
         #region IVsProject Members
 
         public int AddItem(uint itemidLoc, VSADDITEMOPERATION dwAddItemOperation, string pszItemName, uint cFilesToOpen, string[] rgpszFilesToOpen, IntPtr hwndDlgOwner, VSADDRESULT[] pResult) {
-            if (_innerProject3 != null && IsJavaScriptFile(pszItemName)) {
-                Guid ourEditor = typeof(NodejsEditorFactory).GUID;
+            if (_innerProject3 != null && IsJavaScriptFile(pszItemName)) {
+                Guid ourEditor = this.IsES6IntellisensePreview() ? Guid.Empty : typeof(NodejsEditorFactory).GUID;
```

Comment:
How does returning Guid.Empty tell it to use the TypeScript language service for JavaScript files (as opposed to just the regular JSLS editor for JavaScript files)?